src/js/migrations.js

```diff
@@ -261,6 +261,44 @@ exports.Migrations= {
     }
   },
 
+  forgetFirstPartySnitches: function (badger) {
+    console.log("Removing first parties from snitch map...");
+    let snitchMap = badger.storage.getBadgerStorageObject("snitch_map"),
+      actionMap = badger.storage.getBadgerStorageObject("action_map"),
+      snitchClones = snitchMap.getItemClones(),
+      actionClones = actionMap.getItemClones();
+
+    for (let domain in snitchClones) {
+      // creates new array of domains checking against the isThirdParty utility
+      let newSnitches = snitchClones[domain].filter(item => utils.isThirdPartyDomain(item, domain));
+
+      // if that list has changed, reassign or delete the entry as necessary
```

Comment:
This works, but we can simplify by creating a new, corrected `snitchMapClones` instead of modifying (and then restoring) snitch map directly. Something like the following?

```diff
diff --git a/src/js/migrations.js b/src/js/migrations.js
index 854d89a..035a07e 100644
--- a/src/js/migrations.js
+++ b/src/js/migrations.js
@@ -263,35 +263,29 @@ exports.Migrations= {
 
   forgetFirstPartySnitches: function (badger) {
     console.log("Removing first parties from snitch map...");
+
     let snitchMap = badger.storage.getBadgerStorageObject("snitch_map"),
       actionMap = badger.storage.getBadgerStorageObject("action_map"),
       snitchClones = snitchMap.getItemClones(),
-      actionClones = actionMap.getItemClones();
+      actionClones = actionMap.getItemClones(),
+      correctedSites = {};
 
     for (let domain in snitchClones) {
       // creates new array of domains checking against the isThirdParty utility
-      let newSnitches = snitchClones[domain].filter(item => utils.isThirdPartyDomain(item, domain));
+      let newSnitches = snitchClones[domain].filter(
+        item => utils.isThirdPartyDomain(item, domain));
 
-      // if that list has changed, reassign or delete the entry as necessary
-      if (newSnitches.length === 0) {
-        console.log('removing : ', domain);
-        snitchMap.deleteItem(domain);
-        continue;
-      }
-      if (newSnitches.length !== snitchClones[domain].length) {
-        console.log('reassigning : ', domain);
-        snitchMap.setItem(domain, newSnitches);
+      if (newSnitches.length) {
+        correctedSites[domain] = newSnitches;
       }
     }
 
-    // reset the snitch map clones object to the updated snitch map
-    snitchClones = snitchMap.getItemClones();
-
+    // clear existing maps and then use mergeUserData to rebuild them
     actionMap.updateObject({});
     snitchMap.updateObject({});
 
     const data = {
-      snitch_map: snitchClones,
+      snitch_map: correctedSites,
       action_map: actionClones
     };
```

Reply:
Nice -- I like this simplification! Much more readable and reduces some clutter